source/models/row.py

```diff
@@ -14,32 +14,29 @@ def get_primary_key(self):
 
     def serialize(self):
         self.__lock()
-        self.__colomns_name_validate()
+        self.colomns_name_validate(self.get_value())
         row_json_data = json.dumps(self.__value, indent = 2)
-        with open(self.get_path(), 'w') as row_file: 
+        with open(self.get_path(self.get_primary_key()), 'w') as row_file: 
             row_file.write(row_json_data)
         for index_name in self.table.table_metadata.index_keys:
- 
             index = self.table.table_metadata.indicies[index_name]
             index.add_primary_key(primary_key = self.get_primary_key(), index_value = self.__value[index_name])
         self.__unlock()
 
     def row_exists(self):
-        return os.path.isfile(self.get_path())
+        return os.path.isfile(self.get_path(self.get_primary_key()))
```

Comment:
why do you need to do this ?

Reply:
because when joseph implemented this function wasn't abstracted as in side the function he use the primary_key of the value which he want to set . but get and delete function don't have a value it has a query so I change it to take the primarykey parameter

Reply:
so (get path) take the primary key and return the path doesn't it business to know from where the primary key has gotten

Reply:
I answered above, please take a look